### PR TITLE
PP-607/RelayClient: _calculateGasPrice

### PR DESCRIPTION
**Pull request by franciscotobar**
## What

- Add relay client
- Re implement _calculateGasPrice

## Refs

- [PP-605](https://rsklabs.atlassian.net/browse/PP-605)
- https://github.com/rsksmart/rif-relay-common/pull/48

### CODE REVIEW

File: src/RelayClient.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import config from 'config';
+
+class RelayClient {
+  private readonly _provider: providers.Provider;
+
+  private readonly _envelopingConfig: EnvelopingConfig;
+
+  constructor() {
+    this._provider = getDefaultProvider();
+    this._envelopingConfig = config.get('EnvelopingConfig');
```
Comment (AndresQuijano) [inline src/RelayClient.ts:17]: I recommend using a constant for this string.

File: test/RelayClient.test.ts [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+          '_applyGasCorrectionFactor'
+        )
+        .returns(expectedEstimation);
+      const estimateGas = BigNumber.from(10000);
+      provider.estimateGas.returns(Promise.resolve(estimateGas));
+      const internalGasCorrection = 5000;
+      const estimation = await relayClient.estimateInternalCallGas(
+        transactionDetails,
+        internalGasCorrection
+      );
```
Comment (AndresQuijano) [inline test/RelayClient.test.ts:139]: It's a good opportunity to send true as third parameter and test something that hasn't been tested before.

File: test/RelayClient.test.ts [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+      expect(estimation.toString()).to.be.equal(expectedEstimation.toString());
+      expect(stubApplyGasCorrection.calledOnce).to.be.true;
+    });
+
+    /*  it('should truncate internalGasCorrection while doing estimation', async function(){
```
Comment (AndresQuijano) [inline test/RelayClient.test.ts:144]: Something to fix?

**Comment by antomor** [comment]
Closed in favour of #56

**Pull request closed without merging**